APMrover2/sailboat.cpp

```diff
@@ -20,138 +20,212 @@ To Do List
 
     // use PID controller to sheet out
-    const float pid_offset = g2.attitude_control.get_sail_out_from_heel(radians(g2.sail_heel_angle_max), G_Dt) * 100.0f;
+    const float pid_offset =rover.g2.attitude_control.get_sail_out_from_heel(radians(sail_heel_angle_max), rover.G_Dt) * 100.0f;
 
     mainsail = constrain_float((mainsail+pid_offset), 0.0f ,100.0f);
-    g2.motors.set_mainsail(mainsail);
+    rover.g2.motors.set_mainsail(mainsail);
 }
 
 // Velocity Made Good, this is the speed we are traveling towards the desired destination
 // only for logging at this stage
 // https://en.wikipedia.org/wiki/Velocity_made_good
-float Rover::sailboat_get_VMG() const
+float Sailboat::get_VMG() const
 {
     // return 0 if not heading towards waypoint
-    if (!control_mode->is_autopilot_mode()) {
+    if (!rover.control_mode->is_autopilot_mode()) {
         return 0.0f;
     }
 
     float speed;
-    if (!g2.attitude_control.get_forward_speed(speed)) {
+    if (!rover.g2.attitude_control.get_forward_speed(speed)) {
         return 0.0f;
     }
-    return (speed * cosf(wrap_PI(radians(nav_controller->target_bearing_cd()) - ahrs.yaw)));
+    return (speed * cosf(wrap_PI(radians(rover.nav_controller->target_bearing_cd()) - rover.ahrs.yaw)));
 }
 
 // handle user initiated tack while in acro mode
-void Rover::sailboat_handle_tack_request_acro()
+void Sailboat::handle_tack_request_acro()
 {
     // set tacking heading target to the current angle relative to the true wind but on the new tack
-    sailboat.tacking = true;
-    sailboat.tack_heading_rad = wrap_2PI(ahrs.yaw + 2.0f * wrap_PI((g2.windvane.get_absolute_wind_direction_rad() - ahrs.yaw)));
+    currently_tacking = true;
+    tack_heading_rad = wrap_2PI(rover.ahrs.yaw + 2.0f * wrap_PI((rover.g2.windvane.get_absolute_wind_direction_rad() - rover.ahrs.yaw)));
 }
 
 // return target heading in radians when tacking (only used in acro)
-float Rover::sailboat_get_tack_heading_rad() const
+float Sailboat::get_tack_heading_rad() const
 {
-    return sailboat.tack_heading_rad;
+    return tack_heading_rad;
 }
 
 // handle user initiated tack while in autonomous modes (Auto, Guided, RTL, SmartRTL, etc)
-void Rover::sailboat_handle_tack_request_auto()
+void Sailboat::handle_tack_request_auto()
 {
     // record time of request for tack.  This will be processed asynchronously by sailboat_calc_heading
-    sailboat.auto_tack_request_ms = AP_HAL::millis();
+    auto_tack_request_ms = AP_HAL::millis();
 }
 
 // clear tacking state variables
-void Rover::sailboat_clear_tack()
+void Sailboat::clear_tack()
 {
-    sailboat.tacking = false;
-    sailboat.auto_tack_request_ms = 0;
+    currently_tacking = false;
+    auto_tack_request_ms = 0;
 }
 
 // returns true if boat is currently tacking
-bool Rover::sailboat_tacking() const
+bool Sailboat::tacking() const
 {
-    return sailboat.tacking;
+    return currently_tacking && enabled();
```

Comment:
check enable first

Reply:
or not ... currently_tacking is default to false so it will return faster, but we need the check that this clear correctly.
Probably some guy with OCD will complain :-S

Reply:
fixed, thanks